### DISCUSSION TITLE
Code error undefined array key name

### DISCUSSION THREAD

**Discussion by Kiaka3** [bcit-ci/CodeIgniter#6139] [Q&A]
<?php
     session_start();
$y="SELECT * FROM user";
$j=mysqli_query($con,$y);
if($j){
$t=mysqli_fetch_assoc($j);
echo $_SESSION['name'];
}
?>
<h1>WELCOME</h1>
<a href="logout.php">logout</a>
<a href="view.php">view</a>
</form>
</body>
</html>

Error
Warning: Undefined array key "name" in C:\xampp1\htdocs\TEST2\index.php on line 19

**Comment by Oscabrera** [comment]
To better assist you with the error you're encountering, we need more context. Here are some guidelines to help you get more accurate and useful answers:

1. **Provide relevant code**: Include the block of code causing the error. For example, show the contents of the `index.php` file around line 19.
   
   ```php
   <?php
   // Relevant code before line 19
   echo $array['name'];
   // Relevant code after line 19
   ?>
   ```

2. **Problem description**: Explain what you are trying to achieve with the code and what the expected behavior is versus the actual behavior (the error you are seeing).

3. **Environment setup**: Mention details about your development environment, such as the PHP version you are using, the web server (e.g., XAMPP), and any other relevant configurations.

4. **Complete error messages**: Make sure to include the full error message, as you have done, and any other warnings or error messages that appear in your development environment.

For example, your question could look like this:

---

**Original Question:**

Hi, I am getting the following error in my PHP code:

```
Warning: Undefined array key "name" in C:\xampp1\htdocs\TEST2\index.php on line 19
```

**Relevant code:**

```php
<?php
$array = [];
echo $array['name'];
?>
```

I am using PHP 8.1 on XAMPP. How can I fix this error?

---

By following these guidelines, you are more likely to receive a quick and helpful response from the community. Thanks!

---
